src/ape_cache/query.py

```diff
@@ -0,0 +1,331 @@
+from pathlib import Path
+from typing import Any, Dict, Iterator, List, Optional
+
+from hexbytes import HexBytes
+from sqlalchemy import create_engine
+from sqlalchemy.engine import CursorResult  # type: ignore
+from sqlalchemy.sql import column, insert, select, text
+from sqlalchemy.sql.expression import Insert, TextClause
+
+from ape.api import QueryAPI, QueryType
+from ape.api.networks import LOCAL_NETWORK_NAME
+from ape.api.query import BaseInterfaceModel, BlockQuery, BlockTransactionQuery, ContractEventQuery
+from ape.exceptions import QueryEngineError
+from ape.logging import logger
+from ape.utils import singledispatchmethod
+from ape_ethereum.ecosystem import Block
+
+from . import models
+from .models import Blocks, ContractEvents, Transactions
+
+
+class CacheQueryProvider(QueryAPI):
+    """
+    Default implementation of the :class:`~ape.api.query.QueryAPI`.
+    Allows for the query of blockchain data using a connected provider.
+    """
+
+    # Database management
+    def get_database_file(self, ecosystem_name: str, network_name: str) -> Path:
+        """
+        Allows us to figure out what the file *will be*, mostly used for database management.
+        """
+
+        if network_name == LOCAL_NETWORK_NAME:
+            # NOTE: no need to cache local network, no use for data
+            raise QueryEngineError("Cannot cache local data")
```

Comment:
I hit this warning during my initial QA and was a little confused.

I think we need to make the `--network` option required, which we can do.

In `ape.cli.options.py`, we can add a new kwarg `required=False` and pass it along to the click option (or **kwargs).
Then here we can set the value to `True` so you have to provide a value

I think this will lessen the confusion there!

Reply:
This warning message should be different when using the `purge` command.

Right now:

```sh
ape cache purge
ERROR: (QueryEngineError) Cannot cache local data
```

maybe should say something like

```sh
ERROR: (QueryEngineError) No cached data for network 'local'
```

Reply:
Re: the network option adjustment... I am going to try and work that out in a separate PR and will update you then

Reply:
You keep the message if you want, but it seems off to me and maybe we can revise it later.

Reply:
@johnson2427  Thoughts on this: https://github.com/ApeWorX/ape/pull/983/files#r946985076

I think this is the best of both world and then the error message makes the most sense as-is.

Reply:
@unparalleled-js I really like requiring the user to provide a network. It makes the most sense to do that in my opinion. @fubuloubu what do you think? Leave it for now and deal with it later? Or add the requirement now

Reply:
Is required now, we gucci